### DISCUSSION TITLE
How to update keys from patch without adding new keys? (maybe merge_patch)

### DISCUSSION THREAD

**Discussion by JHerrKodo** [nlohmann/json#3391] [Q&A]
Currently merge_patch integrates all keys from patch to the document. I think it might be helpfull, that keys are updated but not added to document. 
Are there any changes planed to add some arguments to merge_patch, or is there any other method which provides this functionality?

**Comment by nlohmann** [comment]
merge_patch() implements RFC 7396 (JSON Merge Patch) literally, and that RFC's algorithm always adds keys from the patch that aren't already in the target - there's no built-in flag to restrict it to 'update existing keys only'. To get that behavior, filter the patch yourself before applying it: recursively walk the patch object and drop any key that doesn't already exist at the corresponding path in the target (keeping nested objects that do exist so their own keys can be filtered the same way), then call target.merge_patch(filtered_patch). There's no planned option to change this since it would deviate from the RFC 7396 semantics that merge_patch() is designed to implement exactly.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
